lib/database/dao/surveys_dao.dart

```diff
@@ -32,41 +31,41 @@ class SurveysDao extends DatabaseAccessor<Database> with _$SurveysDaoMixin {
     return await (select(surveys).get());
   }
 
-  /// Updates persisted Survey by [externalId] and [updatedSurvey]
-  Future<Survey> updateSurveyByExternalId(
-    String externalId,
-    Survey updatedSurvey,
-  ) async {
-    Survey? foundSurvey = await findSurveyByExternalId(externalId);
-
-    if (foundSurvey == null) {
-      logger.shout("Couldn't find Survey with external id $externalId");
-    }
-
-    await update(surveys).replace(foundSurvey!.copyWith(
-      title: updatedSurvey.title,
-      timeout: updatedSurvey.timeout,
-      publishStart: Value(updatedSurvey.publishStart),
-      publishEnd: Value(updatedSurvey.publishEnd),
-    ));
-
-    return await (select(surveys)
-          ..where((row) => row.id.equals(foundSurvey.id)))
-        .getSingle();
-  }
-
   /// Deletes persisted Survey by [id]
   Future deleteSurvey(int id) async {
-    return (delete(surveys)..where((row) => row.id.equals(id)));
+    return await (delete(surveys)..where((row) => row.id.equals(id))).go();
   }
 
   /// Finds currently active Survey
   Future<Survey?> findActiveSurvey() async {
-    return await (select(surveys)
+    Survey? foundSurvey = await (select(surveys)
           ..where((row) =>
               row.publishStart.isSmallerOrEqualValue(DateTime.now()) &
               row.publishEnd.isBiggerOrEqualValue(DateTime.now())))
         .getSingleOrNull();
+
+    foundSurvey ??=
```

Comment:
This returns first one even when the previous query on line 41 should logically return null.